gemd/entity/object/ingredient_run.py

```diff
@@ -53,41 +50,59 @@ class IngredientRun(BaseObject, HasQuantities):
 
     typ = "ingredient_run"
 
-    def __init__(self, material=None, process=None, name=None, labels=None,
-                 mass_fraction=None, volume_fraction=None, number_fraction=None,
-                 absolute_quantity=None,
+    def __init__(self, material=None, process=None, mass_fraction=None,
+                 volume_fraction=None, number_fraction=None, absolute_quantity=None,
                  spec=None, uids=None, tags=None, notes=None, file_links=None):
-        BaseObject.__init__(self, name=name, uids=uids, tags=tags,
+        BaseObject.__init__(self, name=None, uids=uids, tags=tags,
                             notes=notes, file_links=file_links)
-        HasQuantities.__init__(self, mass_fraction, volume_fraction, number_fraction,
-                               absolute_quantity)
-        if name is not None:
-            warnings.warn("The 'name' argument for ingredient runs is deprecated. "
-                          "It may be overwritten by the name of this object's spec.",
-                          DeprecationWarning)
-        if labels is not None:
-            warnings.warn("The 'labels' argument for ingredient runs is deprecated. "
-                          "It may be overwritten by the labels of this object's spec.",
-                          DeprecationWarning)
+        HasQuantities.__init__(self, mass_fraction=mass_fraction, volume_fraction=volume_fraction,
+                               number_fraction=number_fraction, absolute_quantity=absolute_quantity
+                               )
         self._material = None
         self._process = None
         self._spec = None
         self._labels = None
 
         self.material = material
         self.process = process
-        self.labels = labels
-        # this may overwrite name/labels
+        # this will overwrite name/labels if/when they are set
         self.spec = spec
 
+    @property
+    def name(self):
+        """Get name."""
+        return super().name
+
+    @name.setter
+    def name(self, name):
+        # This messiness is a consequence of name being an inherited attribute
+        if name is not None:
+            warnings.warn("Labels are set implicitly by associating with an "
+                          "IngredientSpec; this value will likely be overwritten",
+                          DeprecationWarning)
+        self.__class__._name_setter(self, name)
+
+    @classmethod
+    def _name_setter(cls, obj, name):
+        """Misdirection so assignment in from_dict doesn't hit deprecation warning."""
+        super(IngredientRun, cls).name.fset(obj, name)
```

Comment:
What's this line doing?

Reply:
Dark art that I'm not happy about, but other methods I tried caused problems in libraries that inherited from this.

This expressly sets the name attribute associated with the IngredientRun's parent class (Base Object) without invoking the IngredientRun's overridden setter.  Without this, setting the name via setting an IngredientSpec or deserializing from JSON fires off a deprecation warning every time.